yapapi/strategy.py

```diff
@@ -25,6 +27,30 @@
 class MarketStrategy(DemandDecorator, abc.ABC):
     """Abstract market strategy."""
 
+    valid_prop_value_ranges: Dict[str, Tuple[Optional[float], Optional[float]]]
+
+    def update_valid_prop_value_ranges(
+        self, valid_prop_value_ranges: Dict[str, Tuple[Optional[float], Optional[float]]]
+    ) -> None:
+        try:
+            self.valid_prop_value_ranges.update(valid_prop_value_ranges)
+        except AttributeError:
+            self.valid_prop_value_ranges = valid_prop_value_ranges
```

Comment:
```suggestion
            self.valid_prop_value_ranges = valid_prop_value_ranges.copy()
```
We don't want to modify the received data structure (and neither we want the future modifications of this structure to influence the behavior of the strategy).

Reply:
Fixed